librepo/handle.c

```diff
@@ -841,7 +887,62 @@ lr_handle_setopt(LrHandle *handle,
     return ret;
 }
 
-static gboolean
+gboolean
+lr_handle_network_wait(LrHandle *handle, GError **err, guint seconds, GCancellable *cancellable)
+{
+
+    assert(!err || *err == NULL);
+
+    if (!handle) {
+        g_set_error(err, LR_HANDLE_ERROR, LRE_BADFUNCARG,
+                    "No handle specified");
+        return FALSE;
+    }
+
+    GNetworkMonitor *monitor = g_network_monitor_get_default();
+
+    struct callback_data data_struct;
+    data_struct.cancellable = cancellable;
+    data_struct.monitor = monitor;
+
+    const gchar *baseurl;
+    if(handle->metalinkurl)
+      baseurl = handle->metalinkurl;
+    else if(handle->mirrorlisturl)
+      baseurl = handle->mirrorlisturl;
+    else if(handle->urls)
+      baseurl = handle->urls[0];
+    assert(baseurl);
+
+    g_autoptr(GUri) uri = g_uri_parse(baseurl, G_URI_FLAGS_NONE, NULL);
+    if(uri == NULL){
+      return FALSE;
+    }
+    const gchar* scheme = g_uri_get_scheme(uri);
+    if(!g_strcmp0(scheme, "file")){
```

Comment:
For example, to match the rest of the style this should be
```
    if (!g_strcmp0(scheme, "file")) {
```

Also, be sure you're using 4 spaces and not 2 for indent - across the whole patch.